packages/nimble-components/src/list-option-group/index.ts

```diff
@@ -0,0 +1,121 @@
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { observable, attr, volatile } from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import { ListOption } from '../list-option';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-list-option-group': ListOptionGroup;
+    }
+}
+
+/**
+ * A nimble-styled HTML listbox option
+ */
+export class ListOptionGroup extends FoundationElement {
+    /**
+     * The label for the group.
+     *
+     * @public
+     * @remarks
+     * If a label is also provided via slotted content, the label attribute
+     * will have precedence.
+     */
+    @attr
+    public label?: string;
+
+    /**
+     * The hidden state of the element.
+     *
+     * @public
+     * @defaultValue - false
+     * @remarks
+     * HTML Attribute: hidden
+     */
+    @attr({ mode: 'boolean' })
+    public override hidden = false;
```

Comment:
Is it supported to mark a group `hidden`? If so, can you make sure:
1. the attribute is documented (could happen in the follow up PR where you add a story for the group element)
2. the behavior is well defined. I assume it hides the items under the group too?
3. we have a visual test covering the behavior (could just add a hidden group to the matrix story template)
4. we have unit test coverage as needed. Looks like you already have coverage for filtering of hidden groups but might also want something that asserts setting `hidden` causes `visuallyHidden` to update, for example.

If it's not intended to be a supported API then this should be marked internal.

Reply:
I've added some more tests. Documentation will come in a follow-on PR. Interesting aspect related to this is that I've created a new internal page object class that provides some APIs related to groups.